src/Agent.Plugins/PipelineArtifact/Telemetry/FileShareActionRecord.cs

```diff
@@ -0,0 +1,70 @@
+using System;
+using Agent.Sdk;
+using Microsoft.VisualStudio.Services.Content.Common.Telemetry;
+
+namespace Agent.Plugins.PipelineArtifact.Telemetry
+{
+    /// <summary>
+    /// Generic telemetry record for use with FileShare Artifact events.
+    /// </summary>
+    public class FileShareActionRecord : PipelineArtifactActionRecord
+    {
+        public new long FileCount;
+        public long ContentSize;
+        public long TimeLapse;
+        public string Command;
+        public int ExitCode;
+
+        public FileShareActionRecord(TelemetryInformationLevel level, Uri baseAddress, string eventNamePrefix, string eventNameSuffix, AgentTaskPluginExecutionContext context, uint attemptNumber = 1)
+            : base(level, baseAddress, eventNamePrefix, eventNameSuffix, context, attemptNumber)
+        {
+        }
+
+        protected override void SetMeasuredActionResult<T>(T value)
+        {
+            if (value is FileSharePublishResult)
+            {
+                FileSharePublishResult result = value as FileSharePublishResult;
+                Command = result.Command;
+                TimeLapse = result.TimeLapse;
+                ExitCode = result.ExitCode;
+            }
+
+            if (value is FileShareDownloadResult)
+            {
+                FileShareDownloadResult result = value as FileShareDownloadResult;
+                FileCount = result.FileCount;
+                ContentSize = result.ContentSize;
+                TimeLapse = result.TimeLapse;
+            }
```

Comment:
> } [](start = 11, length = 2)

If the type of value is not of a file share result we should throw InvalidOperationException

Reply:
I don't think telemetry should make the task to fail.